jcache/commons/src/main/java/org/infinispan/jcache/AbstractJCacheNotifier.java

```diff
@@ -57,6 +69,8 @@
 
    private AbstractJCacheListenerAdapter<K,V> listenerAdapter;
 
+   private ConcurrentMap<List<Object>, Queue<CountDownLatch>> latchesByEventSource = new ConcurrentHashMap<>();
```

Comment:
Hmmmm, can you explain this a little? A concurrent map where list is key looks odd, is the key variable in length? If it's always same lenght, can't you take those elements and create a class out of it?